packages/appium/lib/cli/extension-command.js

```diff
@@ -486,4 +487,71 @@ export default class ExtensionCommand {
     this.config.installedExtensions[ext].version = version;
     await this.config.write();
   }
-}
+
+  /**
+   * @typedef RunOutput
+   * @property {string|undefined} error - error message if script ran unsuccessfuly, otherwise undefined
+   * @property {array} output - script output
+   */
+  /**
+   * Runs a script cached inside the "scripts" field under "appium"
+   * inside of the driver/plugins "package.json" file. Will throw
+   * an error if the driver/plugin does not contain a "scripts" field
+   * underneath the "appium" field in its package.json, if the
+   * "scripts" field is not a plain object, or if the scriptName is
+   * not found within "scripts" object.
+   *
+   * @param {string} ext - name of the extension to run a script from
+   * @param {string} scriptName - name of the script to run
+   * @return {RunOutput}
+   */
+  async run ({ext, scriptName}) {
+    if (!_.has(this.config.installedExtensions, ext)) {
+      throw new Error(`please install the ${this.type} first`);
+    }
+
+    const extConfig = this.config.installedExtensions[ext];
+
+    if (!_.has(extConfig, 'scripts')) {
+      throw new Error(`The ${this.type} named '${ext}' does not contain the ` +
+                      `"scripts" field underneath the "appium" field in its package.json`);
+    }
+
+    const extScripts = extConfig.scripts;
+
+    if (!_.isPlainObject(extScripts)) {
+      throw new Error(`"scripts" field must be a plain object`);
+    }
+
+    if (!_.has(extScripts, scriptName)) {
+      throw new Error(`The ${this.type} named '${ext}' does not support the script: '${scriptName}'`);
+    }
+
+    const runner = new SubProcess(`node`, [extScripts[scriptName]],
+                                  {cwd: this.config.getExtensionRequirePath(ext)});
+
+    const output = createRingBuffer(50);
+
+    runner.on('stream-line', (line) => {
+      output.push(line);
+      log(this.isJsonOutput, line);
+    });
+
+    try {
+      // start and wait 5 seconds before throwing an error
```

Comment:
I'm actually not sure whether we need to wait for any output here at all. A script is not obliged to print anything. I would simply use `start(0)` and let it throw when `join` is called